rfcs/text/????-hook-service.md

```diff
@@ -0,0 +1,134 @@
+- `path: string`: The path to the property on the graph service's `blockEntity` property associated with the hook
+- `fallback: (node: HTMLElement, type: "text" | "image" | "video" | string, path: string) => void`: The fallback rendering method which will be used if the embedding application does not implement this particular hook
+- `hookId: string | null`: The ID of the hook as provided in the response when first sending a `hook` message. This should be `null` on first call.  
+
+The embedding application must respond with a `hookResponse` message specifying a `hookId: string` property, which will be provided in future `hook` messages to establish that the hook is simply being updated, and not installed. 
+
+This `hook` message should be sent every time any one of these pieces of information changes (but not, importantly, if the value `path` refers to changes – embedding applications can handle this). In this way, you can think of the message as setting up the hook.
+
+An embedding application receiving a `hook` message should use this to directly manipulate `node` in order to render the `type` view for the specified `path`. Any embedding application implementing the hook service *must* call the `fallback` for any hooks it does not wish to implement. 
+
+When receiving `null` for the `node`, the embedding application should tear down any and all subscriptions made in association with this hook. 
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+TODO:
+
+This is the technical portion of the RFC. Explain the design in sufficient detail that:
+
+- Its interaction with other features is clear.
+- It is reasonably clear how the feature would be implemented.
+- Corner cases are dissected by example.
+
+The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work.
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+- There is the possibility for explosion in values used in `type`, or for blocks to use it to tie their block to a specific embedding application, or set of embedding applications. 
+- The core service which other services build on does not allow sync communication – which may be an issue when it comes to rendering UI. 
+
+# Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+TODO:
+
+- Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not choosing them?
+- What is the impact of not doing this?
+
+# Prior art
+
+[prior-art]: #prior-art
+
+## `editableRef`
+
+[editableRef]: #editableref
+
+The first exploration of this space was the introduction of `editableRef` in the HASH app, currently powering the rich text blocks.
+
+This was designed to avoid blocks being coupled to the particular rich text editing solution implemented in HASH, and in that it succeed.
```

Comment:
```suggestion
This was designed to avoid blocks being coupled to the particular rich text editing solution implemented in HASH, and in that it succeeded.
```